### PR TITLE
feat(query-router): route requests based on schema type

### PR DESCRIPTION
**Pull request by damszew**
Query-router will now:
- using **schema_id** check in `schema-registry`, type of schema and route request either to `query-service` or to `query-service-ts`
- additional parameters (such as range and step needed to query timeseries data) is passed to `query-router` in requests http body

Issue: #15 
BREAKING CHANGE: endpoint `GET /single/<Uuid>` will be changed to HTTP POST and require request body

### CODE REVIEW

File: query-router/src/handler.rs [hunk 1]
```diff
@@ -47,16 +84,30 @@ pub async fn query_multiple(
 
 pub async fn query_by_schema(
     schema_id: Uuid,
-    cache: Arc<AddressCache>,
+    address_cache: Arc<AddressCache>,
+    schema_type_cache: Arc<SchemaTypeCache>,
```
Comment (Kononnable) [inline query-router/src/handler.rs:88]: It looks to me like both caches could be merged - they are used together, fetching both values from schema registry can be done in single request(values are related to same schema_id)

Comment (damszew) [reply query-router/src/handler.rs:88]: Right, in most cases they will be used together

File: query-router/api.yml [hunk 2]
```diff
@@ -22,6 +22,14 @@ paths:
           schema:
             type: string
             example: "15251181-f749-42e0-b4a4-e4b3d90e990d"
+      requestBody:
```
Comment (Kononnable) [inline query-router/api.yml:25]: >Semantic error at paths./single/{object_id}.get.requestBody
GET operations cannot have a requestBody.
Jump to line 25

https://editor.swagger.io/

Comment (damszew) [reply query-router/api.yml:25]: I know but it was decided on meeting with @luke-biel and @esavier that we want to pass additional parameters in request body.
Nevertheless "Request body" part is correctly shown in generated documentation

Comment (luke-biel) [reply query-router/api.yml:25]: @Kononnable how do you see us sending complex requests? I understand it that we need to use GET (we are retrieving data) and queries are complex (now we have 2 params and don't know how many in future)

Comment (Kononnable) [reply query-router/api.yml:25]: And here we go again, tabs or spaces? :smile: 

I know some software use get with request body(e.g. kibana), other uses post requests. If we want to be compatible with OpenAPI Specification we can't use request body in get query - even if RFC 7231 doesn't forbid it. For RFC request body has no known specification(undefined). 
As for using GET - I don't think it's the only option. GET responses can be cached and guarantees idempotency by definition(specification). POST doesn't do that but id doesn't say it's not idempotent either - according to specs post can change data but it's not required to. That means that POST can be used in such cases(however results won't be cached automatically).

TL;DR: I would use POST method here, it's ok with both specs and using get with request body will block us if we want to introduce any form of automatic checks on this api spec file.

Comment (damszew) [reply query-router/api.yml:25]: All right, I've changed this endpoint to POST

**Review by Kononnable — APPROVED** [review]
